unlock-app/src/components/creator/CreatorLock.js

```diff
@@ -0,0 +1,188 @@
+import React from 'react'
+import styled from 'styled-components'
+import UnlockPropTypes from '../../propTypes'
+import Duration from '../helpers/Duration'
+import Balance from '../helpers/Balance'
+import Icon from '../lock/Icon'
+
+import Icons from '../interface/icons'
+
+export function CreatorLock({ lock, status = 'deployed' }) {
+  // Some sanitization of strings to display
+  let name = lock.name || 'New Lock'
+  let displayAddress = lock.address ? lock.address.substr(0, 16) : ''
+  let outstandingKeys = lock.maxNumberOfKeys - lock.outstandingKeys || 0
+
+  if (status === 'deployed') { // the transaction was mined and confirmed at least 12 times
+    // TODO add USD values to lock
+    // TODO add all-time balance to lock
+    return (
+      <CreatorLockRow>
```

Comment:
Would it be worth splitting that in multiple files eventually? All importing the same elements from below?

Reply:
Also, not sure it is worth prefixing `Creator` on all of these components?

Reply:
I wanted to keep the namespace clear for user-facing components, as these are all creator-facing UI elements. But I could change some of the names?

Reply:
Could totally split this up into multiple files though.

Reply:
I'll do exactly that in a follow-up PR.

Reply:
About the naming, I think one of the great thing about the styled components approach is that everything is scoped basically. Thesr components should but be reused.